Services/MovieDataService.swift

```diff
@@ -13,61 +13,87 @@ protocol MovieDataServiceProtocol {
     func getMostPopularMoviesList(completion: @escaping (Result<[Movie], MovieServiceError>)-> Void)
 }
 
-enum MovieServiceError: Error {
+enum MovieServiceError: Error, LocalizedError {
     case failedToGetResponse
     case failedToGetData
-    case failedToDecodeData
-    case errorOccurred
+    case failedToDecode
+    
+    var errorDescription: String? {
+        switch self {
+        case .failedToDecode:
+            return "Failed to decode API response"
+        case .failedToGetResponse:
+            return "Failed to get API response"
+        case .failedToGetData:
+            return "Failed to load data"
+        }
+    }
 }
 
 class MovieDataService: MovieDataServiceProtocol {
+    var playingNowPage = 1
+    var mostPopularPage = 1
+    var isPlayingNowRequestCompleted = true
+    var isMostPopularRequestCompleted = true
```

Comment:
Let's make it private. We don't want other classes messing around with these properties 😁
```suggestion
    private var playingNowPage = 1
    private var mostPopularPage = 1
    private var isPlayingNowRequestCompleted = true
    private var isMostPopularRequestCompleted = true
```